src/graphics/program-lib/chunks/shadowStandard.frag

```diff
@@ -180,3 +180,116 @@ float _getShadowPoint(samplerCube shadowMap, vec4 shadowParams, vec3 dir) {
 float getShadowPointPCF3x3(samplerCube shadowMap, vec4 shadowParams) {
     return _getShadowPoint(shadowMap, shadowParams, dLightDirW);
 }
+
+// ----- Clustered Omni Sampling using atlas -----
+
+// Converts unnormaized direction vector to a cubemap face index [0..5] and uv coordinates within the face in [0..1] range.
+// Additionally offset to a tile in atlas within 3x3 subdivision is provided
+vec2 getCubemapFaceCoordinates(const vec3 dir, out float faceIndex, out vec2 tileOffset)
+{
+	vec3 vAbs = abs(dir);
+	float ma;
+	vec2 uv;
+	if (vAbs.z >= vAbs.x && vAbs.z >= vAbs.y) {   // front / back
+
+		faceIndex = dir.z < 0.0 ? 5.0 : 4.0;
+		ma = 0.5 / vAbs.z;
+		uv = vec2(dir.z < 0.0 ? -dir.x : dir.x, -dir.y);
+
+        tileOffset.x = 2.0;
+        tileOffset.y = dir.z < 0.0 ? 1.0 : 0.0;
+
+	} else if(vAbs.y >= vAbs.x) {  // top index 2, bottom index 3
+
+		faceIndex = dir.y < 0.0 ? 3.0 : 2.0;
+		ma = 0.5 / vAbs.y;
+		uv = vec2(dir.x, dir.y < 0.0 ? -dir.z : dir.z);
+
+        tileOffset.x = 1.0;
+        tileOffset.y = dir.y < 0.0 ? 1.0 : 0.0;
+
+	} else {    // left / right
+
+		faceIndex = dir.x < 0.0 ? 1.0 : 0.0;
+		ma = 0.5 / vAbs.x;
+		uv = vec2(dir.x < 0.0 ? dir.z : -dir.z, -dir.y);
+
+        tileOffset.x = 0.0;
+        tileOffset.y = dir.x < 0.0 ? 1.0 : 0.0;
+
+	}
+	return uv * ma + 0.5;
```

Comment:
Looks like a mixture of tabs and spaces have crept in here. I guess we've standardized on spaces?

Reply:
done, thanks